7 tests fail on i586 architacture

Hi, thank you for this tool! Anyway, I was trying to package it for openSUSE and ran into error on `i586` architecture (32 bit), other (64 bit) architectures are fine. The main theme looks like:
```
[   21s] >       sys.setswitchinterval(original_switch)
[   21s] E       ValueError: switch interval must be strictly positive
[   21s] 
[   21s] /home/abuild/rpmbuild/BUILD/python-pytest-run-parallel-0.4.2-build/BUILDROOT/usr/lib/python3.11/site-packages/pytest_run_parallel/plugin.py:113: ValueError
[   21s] ************************** pytest-run-parallel report **************************
[   21s] 3 tests were not run in parallel because of use of thread-unsafe functionality, to list the tests that were skipped, re-run while setting PYTEST_RUN_PARALLEL_VERBOSE=1 in your shell environment
[   21s] =========================== short test summary info ============================
[   21s] PARALLEL FAILED test_thread_unsafe_function_attr.py::test_should_not_be_marked
[   21s] ========================== 3 passed, 1 error in 0.02s ==========================
```
Could you please take a look into it? The full pytest failure log is in the attachment (too long to paste here).

[log.txt](https://github.com/user-attachments/files/19892370/log.txt)

--- a/src/pytest_run_parallel/plugin.py
+++ b/src/pytest_run_parallel/plugin.py
@@ -65,52 +65,55 @@ def pytest_configure(config):
 
 
 def wrap_function_parallel(fn, n_workers, n_iterations):
-    barrier = threading.Barrier(n_workers)
-    original_switch = sys.getswitchinterval()
-    sys.setswitchinterval(0.000001)
-
     @functools.wraps(fn)
     def inner(*args, **kwargs):
         errors = []
         skip = None
         failed = None
-
-        def closure(*args, **kwargs):
-            for _ in range(n_iterations):
-                barrier.wait()
-                try:
-                    fn(*args, **kwargs)
-                except Warning:
-                    pass
-                except Exception as e:
-                    errors.append(e)
-                except _pytest.outcomes.Skipped as s:
-                    nonlocal skip
-                    skip = s.msg
-                except _pytest.outcomes.Failed as f:
-                    nonlocal failed
-                    failed = f
-
-        workers = []
-        for _ in range(0, n_workers):
-            worker_kwargs = kwargs
-            workers.append(
-                threading.Thread(target=closure, args=args, kwargs=worker_kwargs)
-            )
-
-        num_completed = 0
+        barrier = threading.Barrier(n_workers)
+        original_switch = sys.getswitchinterval()
         try:
+            sys.setswitchinterval(0.000001)
+
+            def closure(*args, **kwargs):
+                for _ in range(n_iterations):
+                    barrier.wait()
+                    try:
+                        fn(*args, **kwargs)
+                    except Warning:
+                        pass
+                    except Exception as e:
+                        errors.append(e)
+                    except _pytest.outcomes.Skipped as s:
+                        nonlocal skip
+                        skip = s.msg
+                    except _pytest.outcomes.Failed as f:
+                        nonlocal failed
+                        failed = f
+
+            workers = []
+            for _ in range(0, n_workers):
+                worker_kwargs = kwargs
+                workers.append(
+                    threading.Thread(
+                        target=closure, args=args, kwargs=worker_kwargs
+                    )
+                )
+
+            num_completed = 0
+            try:
+                for worker in workers:
+                    worker.start()
+                    num_completed += 1
+            finally:
+                if num_completed < len(workers):
+                    barrier.abort()
+
             for worker in workers:
-                worker.start()
-                num_completed += 1
+                worker.join()
+
         finally:
-            if num_completed < len(workers):
-                barrier.abort()
-
-        for worker in workers:
-            worker.join()
-
-        sys.setswitchinterval(original_switch)
+            sys.setswitchinterval(original_switch)
 
         if skip is not None:
             pytest.skip(skip)

--- a/src/pytest_run_parallel/plugin.py
+++ b/src/pytest_run_parallel/plugin.py
@@ -95,9 +95,7 @@ def wrap_function_parallel(fn, n_workers, n_iterations):
             for _ in range(0, n_workers):
                 worker_kwargs = kwargs
                 workers.append(
-                    threading.Thread(
-                        target=closure, args=args, kwargs=worker_kwargs
-                    )
+                    threading.Thread(target=closure, args=args, kwargs=worker_kwargs)
                 )
 
             num_completed = 0

